guardian/front-pressed-lambda
# Fix build

Co-authored-by: @twrichards 



## What does this change?

This PR does a couple of things to touch up a project that hasn't been worked on since 2020. Ultimately, the intention is to get it back in a shape that teamcity can build it and AWS can run it, so that we can then migrate that process to GHA. 

Firstly, this replaces the `get-secrets.sh` logic which was used to access a config file at build time and bake into the bundle; instead the handler function of the lambda will fetch the config from the distbucket specified in the cloudformation. This config then gets passed down to subsequent functions which require various properties specified in it. 

This also bumps the node runtime to 16, which was the most up-to-date version that could be used without having to potentially make significant changes to how the lambda functions. 

## How to test

The most that can be done locally is building the project (`npm run bundle`) and running the suite of tests in index.test.js; one test has long been skipped but the others should pass. 

This has been deployed to code, and we can see the lambda running. 

## How can we measure success?

Teamcity is able to build the project, which riff-raff will then successfully deploy to prod. At that point, we should be able to see usage stats in AWS, and keep being alerted when we should be. 

## Images

Confirmation the lambda has been successfully running since the afternoon of Thursday 19th. 

![image](https://github.com/guardian/front-pressed-lambda/assets/102960844/5314e55b-3b22-4782-a515-7ae78f1008af)
